src/app/static/src/app/components/downloadResults/DownloadResults.vue

```diff
@@ -5,31 +5,28 @@
                 <div id="spectrum-download">
                     <download :translate-key="translation.spectrum"
                               @click="downloadSpectrumOutput"
-                              :disabled="uploading"
-                              :modal-open="uploadModalOpen"
+                              :disabled="!spectrum.downloadId || spectrum.preparing"
                               :file="spectrum"/>
                 </div>
                 <div id="coarse-output-download">
                     <download :translate-key="translation.coarse"
                               @click="downloadCoarseOutput"
-                              :disabled="uploading"
-                              :modal-open="uploadModalOpen"
+                              :disabled="!coarseOutput.downloadId || coarseOutput.preparing"
                               :file="coarseOutput"/>
                 </div>
                 <div id="summary-download">
                     <download :translate-key="translation.summary"
                               @click="downloadSummaryReport"
-                              :disabled="uploading"
-                              :modal-open="uploadModalOpen"
+                              :disabled="!summary.downloadId || summary.preparing"
                               :file="summary"/>
                 </div>
             </div>
             <div id="upload" v-if="hasUploadPermission" class="col-sm">
                 <h4 v-translate="'uploadFileToAdr'"></h4>
                 <button @click.prevent="handleUploadModal"
                         class="btn btn-lg my-3"
-                        :class="uploading || isDownloading ? 'btn-secondary' : 'btn-red'"
-                        :disabled="uploading || isDownloading">
+                        :class="uploading || isPreparing ? 'btn-secondary' : 'btn-red'"
+                        :disabled="uploading || isPreparing">
```

Comment:
I guess technically we only need the spectrum and summary files to be available since coarse outputs aren't uploaded to ADR.